pykafka/cluster.py

```diff
@@ -234,10 +238,14 @@ def _request_metadata(self, broker_connects, topics):
                                     self._offsets_channel_socket_timeout_ms,
                                     buffer_size=1024 * 1024,
                                     source_host=self._source_host,
-                                    source_port=self._source_port)
+                                    source_port=self._source_port,
+                                    ssl_config=self._ssl_config)
                     response = broker.request_metadata(topics)
                     if response is not None:
                         return response
+                # FIXME: handle ssl errors separately somehow, because if we
+                # don't, _get_metadata() might attempt to treat it as if it
+                # were a zookeeper host, which gets messy
```

Comment:
Curious - what happens on the fallback that gets so messy? If it was bad enough it would be a case for permanently [removing](https://github.com/Parsely/pykafka/pull/452) that fallback (which, by the way, I'm not actually certain warrants a major version bump, all things considered).